### PR TITLE
Organization converter R4ToDstu2

### PR DESCRIPTION
**Pull request by bcastillomartel-va**

### CODE REVIEW

File: fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/OrganizationConverter.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                    .address(AddressConverter.INSTANCE.convert(contact.address()))
+                    .purpose(CodeableConceptConverter.INSTANCE.convert(contact.purpose()))
+//                    .extension(
+//                        Safe.stream(contact.extension())
+//                            .map(ExtensionConverter.INSTANCE::convert)
+//                            .toList())
+//                    .modifierExtension(
+//                        Safe.stream(contact.modifierExtension())
+//                            .map(ExtensionConverter.INSTANCE::convert)
+//                            .toList())
```
Comment (jhulbert-va) [inline fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/OrganizationConverter.java:55]: Can be deleted.

File: fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/OrganizationConverter.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  @Mapping(target = "contact", qualifiedByName = "convertContact")
+  @Mapping(target = "type", qualifiedByName = "firstElementOrDie")
+  @Mapping(target = "contained", ignore = true)
+  public abstract gov.va.api.health.dstu2.api.resources.Organization convert(
+      gov.va.api.health.r4.api.resources.Organization source);
+
+  @Named("convertContact")
+  List<gov.va.api.health.dstu2.api.resources.Organization.OrganizationContact> convertContact(
+      List<gov.va.api.health.r4.api.resources.Organization.Contact> source) {
```
Comment (jhulbert-va) [inline fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/OrganizationConverter.java:36]: If you build a Contact to OrganizationContact converter, MapStruct will generate the list one for you. I wouldn't deal with managing lists if you don't have to.

Comment (bcastillomartel-va) [reply fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/OrganizationConverter.java:36]: that's very helpful. Would my test be able to still pass a list of organizationContacts and use `convertContact`?

File: fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/OrganizationConverter.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    return gov.va.api.health.dstu2.api.resources.Organization.OrganizationContact.builder()
+        .id(source.id())
+        .name(HumanNameConverter.INSTANCE.convert(source.name()))
+        .address(AddressConverter.INSTANCE.convert(source.address()))
+        .purpose(CodeableConceptConverter.INSTANCE.convert(source.purpose()))
+        .extension(extensions.isEmpty() ? null : extensions)
+        .modifierExtension(modifierExtensions.isEmpty() ? null : modifierExtensions)
+        .telecom(telecom.isEmpty() ? null : telecom)
+        .build();
+  }
```
Comment (jhulbert-va) [inline fhir-resource-conversion/src/main/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/OrganizationConverter.java:54]: I was talking with someone. was there an error that made you create this method? Based on names, it seems like mapstruct should have generated this code? You may need to just define the abstract method.

**Review by jhulbert-va — DISMISSED** [review]

File: fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/OrganizationSamples.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+  public static R4 r4() {
+    return new R4();
+  }
+
+  public static class Dstu2 {
+    public gov.va.api.health.dstu2.api.resources.Organization organization() {
+      return Organization.builder()
+          .id("1234:I")
+          .telecom(
+              List.of(
```
Comment (jblefkove-va) [inline fhir-resource-conversion/src/test/java/gov/va/api/health/fhir/conversion/r4todstu2/resources/OrganizationSamples.java:24]: There are quite a few places in here where `List.of()` can be replaced with `asList()`

**Review by jblefkove-va — APPROVED** [review]

**Pull request merged**